#! /bin/sh -e

# DP: Description: Fix R_IA64_NONE relocs.
# DP: Author: Randolph Chung <tausq@debian.org> / Andreas Schwab <schwab@suse.de>
# DP: Upstream status: Not submitted
# DP: Status Details: Might be just compensating for a binutils bug
# DP: Date: September 03, 2002

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

--- sysdeps/ia64/dl-machine.h~	Sun Mar 24 16:11:20 2002
+++ sysdeps/ia64/dl-machine.h	Sun Mar 24 16:14:21 2002
@@ -546,12 +546,11 @@
 # endif
         value += map->l_addr;
     }
-# ifndef RTLD_BOOTSTRAP
-  else if (r_type == R_IA64_NONE)
-    return;
-# endif
   else
 #endif
+  if (r_type == R_IA64_NONE)
+    return;
+  else
     {
       struct link_map *sym_map;
